### PR TITLE
Fix socket_type confuse

### PR DESCRIPTION
**Pull request by dark-lbp**
Fix socket_type confuse ql_socket.open should use host platform based socket_type.

### Which kind of PR do you create?

- [x] This PR only contains minor fixes.
- [ ] This PR contains major feature update.
- [ ] This PR introduces a new function/api for Qiling Framework.

### Coding convention?

- [x] The new code conforms to Qiling Framework naming convention.
- [x] The imports are arranged properly.
- [x] Essential comments are added.
- [x] The reference of the new code is pointed out.

### Extra tests?

- [ ] No extra tests are needed for this PR.
- [ ] I have added enough tests for this PR.
- [x] Tests will be added after some discussion and review.

### Changelog?

- [ ] This PR doesn't need to update Changelog.
- [x] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

### CODE REVIEW

File: qiling/os/posix/syscall/socket.py [hunk 1]
```diff
@@ -78,12 +78,24 @@ def ql_syscall_socket(ql, socket_domain, socket_type, socket_protocol, *args, **
...
-            if ql.verbose >= QL_VERBOSE.DEBUG: # set REUSEADDR options under debug mode
+            # ql_socket.open should use host platform based socket_type.
+            try:
+                emu_socket_value = socket_type
+                emu_socket_type = socket_type_mapping(socket_type, ql.archtype)
+                socket_type = getattr(socket, emu_socket_type)
+                ql.log.info("Convert emu_socket_type {}:{} to host platform based socket_type {}:{}".format(
+                    emu_socket_type, emu_socket_value, emu_socket_type, socket_type))
+
+            except Exception:
```
Comment (xwings) [inline qiling/os/posix/syscall/socket.py:89]: I think it must convert.
even a x8664 linux to x8664linux, so we should remove try and except

Comment (dark-lbp) [reply qiling/os/posix/syscall/socket.py:89]: I use this try and except to capture `getattr` exception. Maybe change it like below?
```python
           try:
                emu_socket_value = socket_type
                emu_socket_type = socket_type_mapping(socket_type, ql.archtype)
                socket_type = getattr(socket, emu_socket_type)
                ql.log.info("Convert emu_socket_type {}:{} to host platform based socket_type {}:{}".format(
                    emu_socket_type, emu_socket_value, emu_socket_type, socket_type))

            except AttributeError:
                ql.log.error("Can't convert emu_socket_type {}:{} to host platform based socket_type".format(
                    emu_socket_type, emu_socket_value))
                raise

            except Exception:
                ql.log.error("Can't convert emu_socket_type {} to host platform based socket_type".format(
                    emu_socket_value))
                raise
```

Comment (xwings) [reply qiling/os/posix/syscall/socket.py:89]: agree. we should raise an error here.

**Comment by dark-lbp** [comment]
@xwings  This is more complicate than i thought, maybe we should just log an error and passthrough the convert field value. 
About failed test case socket_type 526338 is SOCK_DGRAM, And i don't know why.
```python 
In [16]: s=socket.socket(2,526338,0)

In [17]: s
Out[17]: <socket.socket fd=16, family=AddressFamily.AF_INET, type=SocketKind.SOCK_DGRAM, proto=0, laddr=('0.0.0.0', 0)>

In [18]: socket.SocketKind.SOCK_DGRAM
Out[18]: <SocketKind.SOCK_DGRAM: 2>

```

**Comment by xwings** [comment]
It we "ignore" the socket. it will still not gonna work anyway?

**Comment by dark-lbp** [comment]
> It we "ignore" the socket. it will still not gonna work anyway?

It will work If the value is correct, this mean we haven't map this value in const.py, otherwise I think setsockopt or ql_socket.open will raise an exception? 
Anyway we can get the value from error log.

**Comment by dark-lbp** [comment]
Fixed, socket_type has a mask, we should and SOCK_TYPE_MASK before mapping.
https://code.woboq.org/linux/linux/net/socket.c.html#1363

**Comment by dark-lbp** [comment]
We need more test.

Tested firmware:

1. D-link DIR816  MIPS Little endian
2. Netgear r8300 ARM Little endian
3. Tenda AC15 ARM Little endian

**Comment by dark-lbp** [comment]
@xwings Can you take a look at this commit? I am not sure why CI failed at `test_elf_linux_execve_x8664 (__main__.ELFTest)`

**Comment by xwings** [comment]
try to rebase your code and try again ?

**Comment by dark-lbp** [comment]
Open a new PR #880 and close this one.

**Pull request closed without merging**